# Домашнє завдання до теми "Python Basics"
В цьому ДЗ важливою умовою виконання завдань є те, що ми не імпортуємо жодних бібліотек, а робимо все на чистому Python, використовуючи лише базовий функціонал. Це необхідно аби відпрацювати роботу з базовими типами даних. А використовувати бібліотеки ми ще навчимось далі :)

## Робота з числами та арифметичними операторами

1.Створіть змінну, яка буде зберігати кількість яблук. І надайте їй значення 5. Виведіть значення змінної на екран.

In [1]:
apple_sum = 5
print (apple_sum)
display (apple_sum)
apple_sum

5


5

5

2.Виведіть тип змінної, створеної в завданні (1).

In [6]:
type (apple_sum) # або інший варіант
#print (type(apple_sum))
#display (type(apple_sum))

int

3.Додайте до значення змінної, створеної в завданні (1) число `12` і виведіть результат на екран з допомогою команди `print()`.

In [7]:
apple_sum += 12
print(apple_sum)

17


4.Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [9]:
a = 101
b = 210
c = 111
d = 402
((a + b) * c) / d

85.8731343283582

Очікувана відповідь: 85.8731343283582

5.Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [11]:
11283 % 89

69

6.Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [12]:
11283 // 89

126

# Робота з рядками

7.Запишіть у змінну `name` ваше імʼя.

In [14]:
name = 'Victoria'

8.Запишіть у змінну `surname` ваше прізвище.

In [15]:
surname = 'Verbytska'

9.Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [16]:
name_surname = name + ' ' +surname
print(name_surname)

Victoria Verbytska


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva' 

10.Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [20]:
print(name_surname.upper())

VICTORIA VERBYTSKA


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

11.Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [19]:
print(name_surname[2])

c


Очікувана відповідь (тільки з вашими даними): 'n'

12.Виведіть першу та останню літеру вашого прізвища.

In [29]:
print( "First surname's letter is ", surname[0]) 
print( "Last surname's letter is ", surname[-1])

First surname's letter is  V
Last surname's letter is  a


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

13.Виведіть довжину вашого прізвища.

In [30]:
len(surname)

9

Очікувана відповідь (тільки з вашими даними): 7

14.Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [33]:
capital = 'Antananarivo'
capital = capital.lower ()
#print ('The capital of Madagascar is ', capital)
capital.count('a')

4

Очікувана відповідь: 4

15.Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [50]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
if my_string.startswith('The'):
    print ("This strins begins with word 'The'")
else: 
    print ("This strins doesn't begins with word 'The'")

if my_string.endswith('dog'):
    print ("This strins ends with word 'dog'")
else:
    print ("This strins doesn't begins with word 'dog'")

This strins begins with word 'The'
This strins doesn't begins with word 'dog'


16.Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [56]:
print (my_string)
my_string_list = my_string.split(' ')
print (my_string_list)
my_string = '_'.join(my_string_list)
print (my_string)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!
['The_quick_brown_fox_jumps_over_the_lazy_dog!!!']
The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`

# Робота зі колекціями

У нас є наступний список:

In [124]:
my_list = [
    78, 12, 14, 27, 47, 28, 51, 46, 94, 37, 42, 20, 52, 33, 11, 11, 51,
    20, 81, 54, 86, 51, 38, 42, 49, 99, 76, 17,  9, 23, 97, 29, 29, 63,
    40, 11, 31, 76, 15, 21, 29, 75, 35,  2, 10, 71, 17, 10, 13,  7, 44,
    12, 19, 61, 13, 25, 62, 46, 10, 44, 75, 57, 20, 57,  3, 44, 16, 28,
    79, 57, 62, 70, 95,  8, 47, 82, 80, 45, 36, 44, 79, 95, 73, 42, 29,
    83, 80, 66, 56,  8, 73,  5, 99, 65, 65, 98, 17, 36, 16, 60
]

Будемо з ним працювати в наступних завданнях.

17.Виведіть 5 перших елементів списку `my_list` з допомогою слайсингу.

In [66]:
print(my_list[:5])

[78, 12, 14, 27, 47]


Очікувана відповідь: [78, 12, 14, 27, 47]

18.Виведіть 3 останніх елементи списку `my_list` з допомогою слайсингу.

In [67]:
print(my_list[-3:])
my_list[-3:]

[36, 16, 60]


[36, 16, 60]

19.Відсортуйте за зростанням список `my_list`, запишіть результат у змінну `my_list_sorted` та виведіть значення нової змінної на екран.

In [89]:
my_list_sorted = sorted(my_list)
print ('My  initial is:',my_list)
print ('')
#my_list.sort()
print ('My sorted list is:', my_list_sorted)

My  initial is: [78, 12, 14, 27, 47, 28, 51, 46, 94, 37, 42, 20, 52, 33, 11, 11, 51, 20, 81, 54, 86, 51, 38, 42, 49, 99, 76, 17, 9, 23, 97, 29, 29, 63, 40, 11, 31, 76, 15, 21, 29, 75, 35, 2, 10, 71, 17, 10, 13, 7, 44, 12, 19, 61, 13, 25, 62, 46, 10, 44, 75, 57, 20, 57, 3, 44, 16, 28, 79, 57, 62, 70, 95, 8, 47, 82, 80, 45, 36, 44, 79, 95, 73, 42, 29, 83, 80, 66, 56, 8, 73, 5, 99, 65, 65, 98, 17, 36, 16, 60]

My sorted list is: [2, 3, 5, 7, 8, 8, 9, 10, 10, 10, 11, 11, 11, 12, 12, 13, 13, 14, 15, 16, 16, 17, 17, 17, 19, 20, 20, 20, 21, 23, 25, 27, 28, 28, 29, 29, 29, 29, 31, 33, 35, 36, 36, 37, 38, 40, 42, 42, 42, 44, 44, 44, 44, 45, 46, 46, 47, 47, 49, 51, 51, 51, 52, 54, 56, 57, 57, 57, 60, 61, 62, 62, 63, 65, 65, 66, 70, 71, 73, 73, 75, 75, 76, 76, 78, 79, 79, 80, 80, 81, 82, 83, 86, 94, 95, 95, 97, 98, 99, 99]


20.Виведіть максимальний і мінімальний елемент у списку `my_list`. Сформуйте відповідь у наступному форматі
```
Max element  <число>
Min element  <число>
```
Можна скористатись результатом з попереднього завдання, або спеціальними фукнціями.

In [93]:
print('''
Variant 1

''')
print ('Max elemant is ', my_list_sorted[-1])
print ('Min elemant is ', my_list_sorted[0])
print(''' 
Variant 2

''')
print ('Max elemant is ', max(my_list))
print ('Min elemant is ', min(my_list))




Variant 1


Max elemant is  99
Min elemant is  2
 
Variant 2


Max elemant is  99
Min elemant is  2


Очікувана відповідь: 

```
Max element  99
Min element  2
```

21.Створіть новий список `my_list_2`, який містить кожен 11-й елемент списку `my_list`. Виведіть `my_list_2` на екран.  
Памʼятаємо, що перший з "кожного одинадцятого елементу" списку має індекс 10.
Для виконання завдання використайте слайсинг.

In [117]:
# print(my_list)
# print(len(my_list))
# c_ount = len(my_list)
my_list_2 = my_list[10:-1:11]
print(my_list_2)

[42, 51, 29, 2, 13, 44, 80, 66, 16]


Очікуваний результат: `[42, 51, 29, 2, 13, 44, 80, 66, 16]`

22.Заданий список `new_list`. Перетворіть його на набір (set). Які зміни у наборі елементів Ви помітили? Відповідь напишіть у коментарі до коду (саме в клітинці з кодом). 

In [128]:
new_list = [22, 11, 67, 11, 88, 3213, 12, 78, 67, 11, 16, 15, 88]

In [140]:
print('''
Variant 1

''')
new_set_1=set(new_list)
print(new_set_1)
print('''

Variant 2

''')
new_set_2 = set()
for x in new_list:
   new_set_2.add(x)
print(new_set_2)

print('''
Variant 3

''')
new_set_3 = {x for x in new_list}
print(new_set_3)


Variant 1


{67, 11, 12, 3213, 78, 15, 16, 22, 88}


Variant 2


{67, 11, 12, 3213, 78, 15, 16, 22, 88}

Variant 3


{67, 11, 12, 3213, 78, 15, 16, 22, 88}


23.Аби отримати скільки разів зустрічається деякий елемент в списку, зручно використовувати метод count(). Працює це наступним чином:
```
some_list = [1,1,1,2,2]
some_list.count(1)
# 3
```

Виведіть, скільки разів зустрічається число 11 у списку `new_list`.

In [141]:
print(new_list.count(11))

3


24.Наостанок, попрактикуємось працювати зі словниками. Це завдання включає 4 невеличких пункти:

1. Створіть словник, що містить 5 пар ключ-значення, де ключі - це імена студентів (типу рядок), а значення - їхні оцінки з програмування від 1 до 12. Запишіть словник у змінну.
2. Виведіть весь словник на екран. Важливо, аби словник вивівся на екран в jupyter notebook, будь-який спосіб, який це робить, нам підходить.
3. Додайте до словника нового студента та його оцінку.
4. Виведіть оцінку одного конкретного студента, звертаючись до неї за його ім'ям.

In [151]:
my_dictionary ={
    'Ivan': 12,
    'Petro': 7,
    'Olena': 11,
    'Stepan': 10,
    'Olga': 9}
print(my_dictionary.items())
my_dictionary['Maria'] = 12
print(my_dictionary.items())
print ('\nCongratulation, your mark is ', my_dictionary['Stepan'])

dict_items([('Ivan', 12), ('Petro', 7), ('Olena', 11), ('Stepan', 10), ('Olga', 9)])
dict_items([('Ivan', 12), ('Petro', 7), ('Olena', 11), ('Stepan', 10), ('Olga', 9), ('Maria', 12)])

Congratulation, your mark is  10


# Ураа!! Ви завершили перше домашнє завдання!!!  
Аби здати ДЗ - залийте ноутбук у гітхаб репозиторій, який ви створили у перші дні навчання на курсі, та додайте посилання на файл у відповідному завданні на навчальній платформі.
Так тримати! 🚀🚀🚀